packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -0,0 +1,73 @@
+import { expect, stub } from './utils';
+import { TemplateResult } from 'lit-element';
+import Sayt from '../src/sayt';
+import { SAYT_EVENT } from '../src/events';
+
+describe('Sayt Component', () => {
+  let sayt;
+
+  beforeEach(() => {
+    sayt = new Sayt();
+  });
+
+  describe('connectedCallback()', () => {
+    it('should register event listeners to the window', () => {
+      const addEventListener = stub(window, 'addEventListener');
+
+      sayt.connectedCallback();
+
+      expect(addEventListener).to.be.calledWith(SAYT_EVENT.SAYT_SHOW, sayt.showSayt);
+      expect(addEventListener).to.be.calledWith(SAYT_EVENT.SAYT_HIDE, sayt.hideSayt);
+    });
+  });
+
+
+  describe('disconnectedCallback()', () => {
+    it('should remove event listeners on the window', () => {
+      const removeEventListener = stub(window, 'removeEventListener');
+
+      sayt.disconnectedCallback();
+
+      expect(removeEventListener).to.be.calledWith(SAYT_EVENT.SAYT_SHOW, sayt.showSayt);
+      expect(removeEventListener).to.be.calledWith(SAYT_EVENT.SAYT_HIDE, sayt.hideSayt);
+    });
+  });
+
+  describe('updated()', () => {
+    it('should change the hidden property if the visible property has changed', () => {
+      sayt.hidden = true;
+      sayt.visible = true;
+
+      sayt.updated(new Map([['visible', true]]));
+
+      expect(sayt.hidden).to.be.false;
+    });
+  });
+
+  describe('showSayt()', () => {
+    it('should set the visible prop to true', () => {
```

Comment:
I feel that this test requires first asserting that `sayt.visible` is `false`. It keeps this test independent and properly tells the story of it becoming `true`. Otherwise, we have to look elsewhere to the defaults to be sure that it wasn't already visible.

Reply:
Updated.